### PR TITLE
HMSPROV-115: Update makefile, add pr_checks and build_deploy

### PR DESCRIPTION
**Pull request by mshriver**
These will be used by appsre CI for building the main branch image and
pushing to quay.io/cloudservices, and for executing PR checks against
all PRs opened in this repo.

pr_checks will call CI scripts from a shared repo to build an image with
the PR content and run iqe tests against it

build_deploy will execute against the main branch after PR merges.

### Testing

Executed build_deploy locally, producing and pushing this image:
https://quay.io/repository/mshriver/provisioning-backend?tab=tags

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,11 +19,15 @@ strip: build
 clean:
 	rm pbapi pbmigrate
 
+
+DOCKER_BUILD_OPTS ?= "--build-arg quay_expiration=2d"
+DOCKER_IMAGE ?= "provisioning-backend"
```
Comment (ezr-ondrej) [inline Makefile:24]: Could we move these up the file with a comment for what they are used?
Lukas wants to split up the make file a bit in the future (in the manner Content has it) but in the meantime I'd like to have Vars at the top.

Comment (ezr-ondrej) [inline Makefile:24]: Also, given we don't build by docker, should we name them `CONTAINER_*`
```suggestion
CONTAINER_BUILD_OPTS  ?= "--build-arg quay_expiration=2d"
CONTAINER_IMAGE_NAME ?= "provisioning-backend"
```

File: build_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    registry.redhat.io
+
+# build and push
+make build-podman \
+    DOCKER_BUILD_OPTS=--no-cache \
+    # TODO opts for image base/tag 
+
+# push to logged in registries and tag for SHA
+docker tag "${IMAGE}:${IMAGE_TAG}" "${IMAGE}:${SMOKE_TEST_TAG}"
+docker push "${IMAGE}:${SMOKE_TEST_TAG}"
```
Comment (ezr-ondrej) [inline build_deploy.sh:37]: can we use podman here? I'd stick with one tool and podman is bit lighter dependency, so we wanted to use podman if at all possible.

Comment (mshriver) [reply build_deploy.sh:37]: Absolutely - thanks for the quick review! Will make all this consistent.

**Review by ezr-ondrej — APPROVED** [review]
Looks ready to me, do we want for it to be useful, or shall we merge? I'm ok with both :)

**Comment by mshriver** [comment]
@ezr-ondrej you saying my PR isn't useful?? :) Doesn't hurt either way, but its not needed until I've got the app-interface followup with the jenkins-config and saas definitions.

Happy to take more review input/suggestions in the meantime.

**Comment by mshriver** [comment]
@ezr-ondrej latest commit means this should be merged after the app-interface PR containing the build and deploy configs. Opening that now.

**Comment by ezr-ondrej** [comment]
Yeah the repos are created, but I guess we do not have permissions to push there... :thinking:

**Comment by mshriver** [comment]
@ezr-ondrej note my last commit changes the clowdapp template name to match the resourceTemplate in the SAAS spec.

**Comment by ezr-ondrej** [comment]
> note my last commit changes the clowdapp template name to match the resourceTemplate in the SAAS spec.

Yeah, those make sense I believe, I'm not sure where should be app name and where component name, so I'll trust you on that one ^_^

**Pull request merged**